library/modules/bigip_wait.py

```diff
@@ -319,11 +334,40 @@ def _is_mprov_running_on_device(self):
             return True
         return False
 
+    def _is_vcmpd_running_on_device(self):
+        params = {
+            "command": "run",
+            "utilCmdArgs": '-c "ps aux | grep \'[v]cmpd\'"'
+        }
+        uri = "https://{0}:{1}/mgmt/tm/util/bash".format(
+            self.client.provider['server'],
+            self.client.provider['server_port']
+        )
+        resp = self.client.api.post(uri, json=params)
+        try:
+            response = resp.json()
+        except ValueError as ex:
+            raise F5ModuleError(str(ex))
+        if 'code' in response and response['code'] in [400, 403]:
+            if 'message' in response:
+                raise F5ModuleError(response['message'])
+            else:
+                raise F5ModuleError(resp.content)
+
+        if 'commandResult' in response:
+            return True
+        return False
+
 
 class ArgumentSpec(object):
     def __init__(self):
         self.supports_check_mode = True
         argument_spec = dict(
+            type=dict(
+                choices=['standard', 'vcmp'],
+                default='standard',
+                type='str'
```

Comment:
no need to define str type in arg_spec remove :)

Reply:
But isn't it good to define it as such?

Reply:
no it is defined explicitly in the documentation, arg_spec is not where it is placed

Reply:
Done.